### PR TITLE
HookStack: Implement version prune in AfterAllowTraffic

### PR DESCRIPTION
**Pull request by AaronMoat**
Resolves https://github.com/seek-oss/aws-codedeploy-hooks/issues/32 and also fixes it being very broken

### CODE REVIEW

File: packages/infra/src/constructs/stack.ts [hunk 1]
```diff
@@ -52,9 +78,9 @@ export class HookStack extends Stack {
     );
 
     // Deny access to resources that have a falsy `aws-codedeploy-hooks` tag.
-    beforeAllowTrafficHook.addToRolePolicy(
+    hookFunction.addToRolePolicy(
       new aws_iam.PolicyStatement({
-        actions: ['*'],
+        actions,
```
Comment (72636c) [inline packages/infra/src/constructs/stack.ts:83]: Any downside to leaving the `*` for these denies?

File: packages/infra/src/handlers/process/lambda/prune.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  logger.info(
+    `Pruning function versions: ${versionsToPrune
+      .map((version) => version.Version)
+      .join(', ')}`,
+  );
+
+  await Promise.all(
+    versionsToPrune.map((version) =>
+      lambda.send(
+        new DeleteFunctionCommand({
```
Comment (72636c) [inline packages/infra/src/handlers/process/lambda/prune.ts:63]: Any potential concerns around this fanning out thousands of parallel requests on first run?

Could rate limiting / timeouts come into play? 😅

**Review by 72636c — APPROVED** [review]
🙇

File: packages/infra/src/constructs/stack.ts (hunk 1 above)
Comment (AaronMoat) [reply packages/infra/src/constructs/stack.ts:83]: It was blocking creation of log groups for the lambda. Any other ideas?

File: packages/infra/src/handlers/process/lambda/prune.ts (hunk 2 above)
Comment (AaronMoat) [reply packages/infra/src/handlers/process/lambda/prune.ts:63]: Yeah I suppose the _deletes_ should be limited

**Pull request merged**